src/Components/Server/src/Circuits/RemoteRenderer.cs

```diff
@@ -112,7 +113,13 @@ protected override Task UpdateDisplayAsync(in RenderBatch batch)
             // snapshot its contents now.
             // TODO: Consider using some kind of array pool instead of allocating a new
             //       buffer on every render.
-            var batchBytes = MessagePackSerializer.Serialize(batch, RenderBatchFormatterResolver.Instance);
+            var memoryStream = new MemoryStream();
+            using (var renderBatchWriter = new RenderBatchWriter(memoryStream, false))
+            {
+                renderBatchWriter.Write(in batch);
+            }
+
+            var batchBytes = memoryStream.ToArray();
```

Comment:
You can avoid the second array allocation if you supported Memory\<byte\> or ArraySegment\<byte\>

Reply:
Would this be in SignalR?